### PR TITLE
client,server: better err msg when PD endpoint missing certificate

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve? 
close https://github.com/pingcap/ticdc/issues/1973

### What is changed and how it works?

Add better err msg.
### Check List 

Tests 

 - Unit test

Code changes

 - Has exported variable/fields change

Side effects

None

Related changes

 - Need to cherry-pick to the release branch

### Release note

- Better err msg when PD endpoint missing certificate

### CODE REVIEW

**Comment by hi-rustin** [comment]
I'm not sure we need a more detailed error message (e.g. prompting the user to use parameters to specify a certificate), suggestions welcome!

**Comment by hi-rustin** [comment]
Also, should we check that it is a valid URL first?

**Review by leoppro — APPROVED** [review]

File: cmd/server.go [hunk 1]
```diff
@@ -204,11 +210,15 @@ func loadAndVerifyServerConfig(cmd *cobra.Command) (*config.ServerConfig, error)
...
 				return nil, cerror.ErrInvalidServerOption.GenWithStack("PD endpoint scheme should be https")
 			}
-		} else if strings.Index(ep, "http://") != 0 {
-			return nil, cerror.ErrInvalidServerOption.GenWithStack("PD endpoint scheme should be http")
+		} else {
+			if strings.HasPrefix(ep, HTTPS) {
+				return nil, errors.New("PD endpoint scheme is https, please provide certificate")
+			} else if !strings.HasPrefix(ep, HTTP) {
+				return nil, cerror.ErrInvalidServerOption.GenWithStack("PD endpoint scheme should be http")
+			}
```
Comment (lonng) [inline cmd/server.go:221]: Seems the same code was duplicated, and one has the error code but the other hasn't.

File: cmd/client.go [hunk 2]
```diff
@@ -138,11 +138,15 @@ func newCliCommand() *cobra.Command {
 				return errors.Annotate(err, "fail to validate TLS settings")
 			}
 			if tlsConfig != nil {
-				if strings.Contains(cliPdAddr, "http://") {
+				if strings.HasPrefix(cliPdAddr, HTTP) {
```
Comment (lonng) [inline cmd/client.go:141]: Why not parse the URL to check the scheme information?

Comment (hi-rustin) [reply cmd/client.go:141]: Yes, it is best to do a URL validate first: https://github.com/pingcap/ticdc/pull/2138#issuecomment-867303878

I will add that step.

File: cmd/server.go (hunk 1 above)
Comment (hi-rustin) [reply cmd/server.go]: Good catch! Fixed.

File: cmd/client.go (hunk 2 above)
Comment (hi-rustin) [reply cmd/client.go]: Added. Could you please take again?

**Comment by hi-rustin** [comment]
/hold 

Code can be better.

**Comment by hi-rustin** [comment]
/unhold

Ready for review.

**Review by overvenus — APPROVED** [review]

**Pull request merged**